mteb/tasks/BitextMining/multilingual/TedTalksBitextMining.py

```diff
@@ -0,0 +1,174 @@
+from __future__ import annotations
+
+from typing import Any
+
+from datasets import DatasetDict, load_dataset
+
+from mteb.abstasks import AbsTaskBitextMining, CrosslingualTask
+from mteb.abstasks.TaskMetadata import TaskMetadata
+
+# Mapping from ISO 639-3 + script to ISO 639-1 used in TED
+_ISO6393_to_ISO6391 = {
+    "ara_Arab": "ar",
+    "aze_Latn": "az",
+    "bel_Cyrl": "be",
+    "bul_Cyrl": "bg",
+    "ben_Beng": "bn",
+    "bos_Latn": "bs",
+    "ces_Latn": "cs",
+    "dan_Latn": "da",
+    "deu_Latn": "de",
+    "ell_Grek": "el",
+    "eng_Latn": "en",
+    "epo_Latn": "eo",
+    "spa_Latn": "es",
+    "est_Latn": "et",
+    "eus_Latn": "eu",
+    "fas_Arab": "fa",
+    "fin_Latn": "fi",
+    "fra_Latn": "fr",
+    "glg_Latn": "gl",
+    "heb_Hebr": "he",
+    "hin_Deva": "hi",
+    "hrv_Latn": "hr",
+    "hun_Latn": "hu",
+    "hye_Armn": "hy",
+    "ind_Latn": "id",
+    "ita_Latn": "it",
+    "jpn_Jpan": "ja",
+    "kat_Geor": "ka",
+    "kaz_Cyrl": "kk",
+    "kor_Hang": "ko",
+    "kur_Arab": "ku",
+    "lit_Latn": "lt",
+    "mkd_Cyrl": "mk",
+    "mon_Cyrl": "mn",
+    "mar_Deva": "mr",
+    "msa_Latn": "ms",
+    "mya_Mymr": "my",
+    "nob_Latn": "nb",
+    "nld_Latn": "nl",
+    "pol_Latn": "pl",
+    "por_Latn": "pt",
+    "ron_Latn": "ro",
+    "rus_Cyrl": "ru",
+    "slk_Latn": "sk",
+    "slv_Latn": "sl",
+    "sqi_Latn": "sq",
+    "srp_Cyrl": "sr",
+    "swe_Latn": "sv",
+    "tam_Taml": "ta",
+    "tha_Thai": "th",
+    "tur_Latn": "tr",
+    "ukr_Cyrl": "uk",
+    "urd_Arab": "ur",
+    "vie_Latn": "vi",
+    "zho_Hans": "zh-cn",
+    "zho_Hant": "zh-tw",
+}
+
+_SPLIT = ["train"]
+
+# number of sentences to use for evaluation
+_N = 256
+
+
+def extend_lang_pairs() -> dict[str, list[str]]:
+    # add all possible language pairs
+    hf_lang_subset2isolang = {}
+    for x in _ISO6393_to_ISO6391.keys():
+        if "-" not in x:
+            for y in _ISO6393_to_ISO6391.keys():
+                if x != y:
+                    pair = f"{x}-{y}"
+                    hf_lang_subset2isolang[pair] = [
+                        x.replace("_", "-"),
+                        y.replace("_", "-"),
+                    ]
+    return hf_lang_subset2isolang
```

Comment:
This function is the same as in [Flores](https://github.com/embeddings-benchmark/mteb/blob/main/mteb/tasks/BitextMining/multilingual/FloresBitextMining.py) right? Maybe we should define it in AbsTaskBitextMining to allow reusing it eaily and not duplicate the code, wdyt?

Reply:
it seems dependent mostly on the label format which is dataset-specific. I would keep it separate as changing one dataset should not lead to changes in another (this makes them independent).

Reply:
@davidstap you can resolve this conversation 🙂